source/extensions/filters/network/dubbo_proxy/router/router_impl.cc

```diff
@@ -0,0 +1,324 @@
+
+  conn_pool_handle_ = nullptr;
+
+  // Mimic an upstream reset.
+  onUpstreamHostSelected(host);
+  onResetStream(reason);
+
+  conn_pool_handle_ = nullptr;
+  parent_.upstream_request_buffer_.drain(parent_.upstream_request_buffer_.length());
+}
+
+void Router::UpstreamRequest::onPoolReady(Tcp::ConnectionPool::ConnectionDataPtr&& conn_data,
+                                          Upstream::HostDescriptionConstSharedPtr host) {
+  ENVOY_LOG(debug, "dubbo upstream request: tcp connection has ready");
+
+  // Only invoke continueDecoding if we'd previously stopped the filter chain.
+  bool continue_decoding = conn_pool_handle_ != nullptr;
+
+  onUpstreamHostSelected(host);
+  conn_data_ = std::move(conn_data);
+  conn_data_->addUpstreamCallbacks(parent_);
+  conn_pool_handle_ = nullptr;
+
+  onRequestStart(continue_decoding);
+}
+
+void Router::UpstreamRequest::onRequestStart(bool continue_decoding) {
+  ENVOY_LOG(debug, "dubbo upstream request: start sending data to the server {}",
+            upstream_host_->address()->asString());
+
+  if (buffered_request_body_) {
+    conn_data_->connection().write(*buffered_request_body_, false);
+    parent_.callbacks_->streamInfo().onLastUpstreamTxByteSent();
+  }
+
+  if (continue_decoding) {
+    parent_.callbacks_->continueDecoding();
+  }
+  onRequestComplete();
+}
+
+void Router::UpstreamRequest::onRequestComplete() { request_complete_ = true; }
+
+void Router::UpstreamRequest::onResponseComplete() {
+  response_complete_ = true;
+  conn_data_.reset();
+}
+
+void Router::UpstreamRequest::onUpstreamHostSelected(Upstream::HostDescriptionConstSharedPtr host) {
+  ENVOY_LOG(debug, "dubbo upstream request: selected upstream {}", host->address()->asString());
+  upstream_host_ = host;
+}
+
+void Router::UpstreamRequest::onResetStream(Tcp::ConnectionPool::PoolFailureReason reason) {
+  if (metadata_->message_type() == MessageType::Oneway) {
+    // For oneway requests, we should not attempt a response. Reset the downstream to signal
+    // an error.
+    ENVOY_LOG(debug, "dubbo upstream request: the request is oneway, reset downstream connection");
+    parent_.callbacks_->resetDownstreamConnection();
+    return;
+  }
+
+  switch (reason) {
+  case Tcp::ConnectionPool::PoolFailureReason::Overflow:
+    parent_.callbacks_->sendLocalReply(
+        AppException(ResponseStatus::ServerError,
+                     fmt::format("dubbo upstream request: too many connections to '{}'",
+                                 upstream_host_->address()->asString())),
+        false);
+    break;
+  case Tcp::ConnectionPool::PoolFailureReason::LocalConnectionFailure:
+    // Should only happen if we closed the connection, due to an error condition, in which case
+    // we've already handled any possible downstream response.
+    parent_.callbacks_->resetDownstreamConnection();
+    break;
+  case Tcp::ConnectionPool::PoolFailureReason::RemoteConnectionFailure:
```

Comment:
If the connection is closed remotely, how should the current ActiveMessage be destructed?

Reply:
If RemoteConnectionFailure mistakes before not yet received the response, will be called sendLocalReply, this function will close the downstream connection, will trigger a ConnectionManager: :onEvent function call, will eventually trigger ActiveMessage: :onReset function calls.
If you have received a response, it will call resetDownstreamConnection, will eventually trigger ActiveMessage: : onReset function calls.

Reply:
Where will close the Downstream connection?  why do you close?

Reply:
done

Reply:
I added a resetStream interface that notified downstream module to release the current stream when the upstream connection was closed，downstream connection will not be closed.